src/openzeppelin/account/library.cairo

```diff
@@ -1,334 +1,248 @@
-            syscall_ptr : felt*,
-            pedersen_ptr : HashBuiltin*,
-            range_check_ptr
-        }(_public_key: felt):
-        Account_public_key.write(_public_key)
-        ERC165.register_interface(IACCOUNT_ID)
-        return()
-    end
-
-    #
-    # Guards
-    #
-
-    func assert_only_self{syscall_ptr : felt*}():
-        let (self) = get_contract_address()
-        let (caller) = get_caller_address()
-        with_attr error_message("Account: caller is not this account"):
-            assert self = caller
-        end
-        return ()
-    end
-
-    #
-    # Getters
-    #
-
-    func get_public_key{
-            syscall_ptr : felt*,
-            pedersen_ptr : HashBuiltin*,
-            range_check_ptr
-        }() -> (res: felt):
-        let (res) = Account_public_key.read()
-        return (res=res)
-    end
-
-    func get_nonce{
-            syscall_ptr : felt*,
-            pedersen_ptr : HashBuiltin*,
-            range_check_ptr
-        }() -> (res: felt):
-        let (res) = Account_current_nonce.read()
-        return (res=res)
-    end
-
-    #
-    # Setters
-    #
-
-    func set_public_key{
-            syscall_ptr : felt*,
-            pedersen_ptr : HashBuiltin*,
-            range_check_ptr
-        }(new_public_key: felt):
-        assert_only_self()
-        Account_public_key.write(new_public_key)
-        return ()
-    end
-
-    #
-    # Business logic
-    #
+func Account_public_key() -> (public_key: felt) {
+}
+
+//
+// Structs
+//
+
+struct Call {
+    to: felt,
+    selector: felt,
+    calldata_len: felt,
+    calldata: felt*,
+}
+
+// Tmp struct introduced while we wait for Cairo
+// to support passing `[AccountCall]` to __execute__
+struct AccountCallArray {
+    to: felt,
+    selector: felt,
+    data_offset: felt,
+    data_len: felt,
+}
+
+namespace Account {
+    //
+    // Initializer
+    //
+
+    func initializer{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        _public_key: felt
+    ) {
+        Account_public_key.write(_public_key);
+        return ();
+    }
+
+    //
+    // Guards
+    //
+
+    func assert_only_self{syscall_ptr: felt*}() {
+        let (self) = get_contract_address();
+        let (caller) = get_caller_address();
+        with_attr error_message("Account: caller is not this account") {
+            assert self = caller;
+        }
+        return ();
+    }
+
+    //
+    // Getters
+    //
+
+    func get_public_key{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (
+        public_key: felt
+    ) {
+        return Account_public_key.read();
+    }
+
+    func supports_interface{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(interface_id: felt) -> (
+        success: felt
+    ) {
+        if (interface_id == IERC165_ID) {
+            return (TRUE,);
+        }
+        if (interface_id == IACCOUNT_ID) {
+            return (TRUE,);
+        }
+        return (FALSE,);
```

Comment:
imo the comma scenario is the worst, i'd rather have the return value binded (`return (success=TRUE)`) than that, open to opinions

Reply:
Haha I don't mind the comma so much. I think the binding can make the code look a bit cluttered, but I'm not married to the style.

Your call, change it everywhere in the repo?

```
            return (succes=TRUE);
        }
        if (interface_id == IACCOUNT_ID) {
            return (success=TRUE);
        }
        return (success=FALSE);
```